app/javascript/app/components/emission-pathway-graph/emission-pathway-graph-selectors.js

```diff
@@ -258,9 +272,110 @@ export const getChartConfig = createSelector(
   }
 );
 
+// Parse Metadata for Modal
+export const getModelSelectedMetadata = createSelector(
+  [getModels, getModelSelected],
+  (models, modelSelected) => {
+    if (!models || !modelSelected) return null;
+    return models.find(m => modelSelected.value === m.id.toString());
+  }
+);
+
+export const addLinktoModelSelectedMetadata = createSelector(
+  [getModelSelectedMetadata],
+  model => {
+    if (!model) return null;
+    return {
+      ...model,
+      Link: `/emission-pathways/models/${model.id}`
+    };
+  }
+);
+
+export const getScenariosSelectedMetadata = createSelector(
+  [getScenarios, getScenariosSelected],
+  (scenarios, scenariosSelected) => {
+    if (isEmpty(scenarios) || !scenariosSelected) return null;
+    const selectedScenarioIds = scenariosSelected.map(s => s.value);
+    const scenariosMetadata = scenarios.filter(
+      s => selectedScenarioIds.indexOf(s.id.toString()) > -1
+    );
+    return (
+      scenariosMetadata &&
```

Comment:
this will fail is the map above returns an empty array. Use length, as filter always returns an array.